src/de/mod.rs

```diff
@@ -819,49 +935,69 @@ mod tests {
         assert_eq!(crate::from_str(r#" "thing" "#), Ok((Type::Thing, 9)));
     }
 
+    #[test]
+    fn char() {
+        assert_eq!(crate::from_str("\"n\""), Ok(('n', 3)));
+        assert_eq!(crate::from_str("\"\\\"\""), Ok(('"', 4)));
+        assert_eq!(crate::from_str("\"\\\\\""), Ok(('\\', 4)));
+        assert_eq!(crate::from_str("\"/\""), Ok(('/', 3)));
+        assert_eq!(crate::from_str("\"\\b\""), Ok(('\x08', 4)));
+        assert_eq!(crate::from_str("\"\\f\""), Ok(('\x0C', 4)));
+        assert_eq!(crate::from_str("\"\\n\""), Ok(('\n', 4)));
+        assert_eq!(crate::from_str("\"\\r\""), Ok(('\r', 4)));
+        assert_eq!(crate::from_str("\"\\t\""), Ok(('\t', 4)));
+        assert_eq!(crate::from_str("\"\\u000b\""), Ok(('\x0B', 8)));
+        assert_eq!(crate::from_str("\"\\u000B\""), Ok(('\x0B', 8)));
+        assert_eq!(crate::from_str("\"\u{3A3}\""), Ok(('\u{3A3}', 4)));
+    }
+
     #[test]
     fn str() {
+        // No escaping, so can borrow from the input
         assert_eq!(crate::from_str(r#" "hello" "#), Ok(("hello", 9)));
         assert_eq!(crate::from_str(r#" "" "#), Ok(("", 4)));
         assert_eq!(crate::from_str(r#" " " "#), Ok((" ", 5)));
         assert_eq!(crate::from_str(r#" "👏" "#), Ok(("👏", 8)));
 
-        // no unescaping is done (as documented as a known issue in lib.rs)
-        assert_eq!(crate::from_str(r#" "hel\tlo" "#), Ok(("hel\\tlo", 11)));
-        assert_eq!(crate::from_str(r#" "hello \\" "#), Ok(("hello \\\\", 12)));
+        fn s(s: &'static str) -> heapless::String<1024> {
+            core::str::FromStr::from_str(s).expect("Failed to create test string")
+        }
 
         // escaped " in the string content
-        assert_eq!(crate::from_str(r#" "foo\"bar" "#), Ok((r#"foo\"bar"#, 12)));
+        assert_eq!(
+            crate::from_str(r#" "foo\"bar" "#),
+            Ok((s(r#"foo"bar"#), 12))
+        );
         assert_eq!(
             crate::from_str(r#" "foo\\\"bar" "#),
-            Ok((r#"foo\\\"bar"#, 14))
+            Ok((s(r#"foo\"bar"#), 14))
         );
         assert_eq!(
             crate::from_str(r#" "foo\"\"bar" "#),
-            Ok((r#"foo\"\"bar"#, 14))
+            Ok((s(r#"foo""bar"#), 14))
         );
-        assert_eq!(crate::from_str(r#" "\"bar" "#), Ok((r#"\"bar"#, 9)));
-        assert_eq!(crate::from_str(r#" "foo\"" "#), Ok((r#"foo\""#, 9)));
-        assert_eq!(crate::from_str(r#" "\"" "#), Ok((r#"\""#, 6)));
+        assert_eq!(crate::from_str(r#" "\"bar" "#), Ok((s(r#""bar"#), 9)));
+        assert_eq!(crate::from_str(r#" "foo\"" "#), Ok((s(r#"foo""#), 9)));
+        assert_eq!(crate::from_str(r#" "\"" "#), Ok((s(r#"""#), 6)));
 
         // non-excaped " preceded by backslashes
         assert_eq!(
             crate::from_str(r#" "foo bar\\" "#),
-            Ok((r#"foo bar\\"#, 13))
+            Ok((s(r#"foo bar\"#), 13))
```

Comment:
All of these tests should be requiring an unescape buffer. Why are they working? Is this because of the implied 128 byte buffer? If that's the case, that's not how we want this to work

Reply:
I've changed `crate::from_str` and `crate::from_slice` to use an empty de-escape buffer, and thus buffer usage in the tests is now more explicit.